frontend/src/pages/acceleratorProfiles/screens/manage/tolerations/ManageTolerationModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import { Modal, Button, Form } from '@patternfly/react-core';
+import { PodToleration } from '~/types';
+import TolerationFields from './TolerationFields';
+import { EMPTY_TOLERATION } from './const';
+
+type ManageTolerationModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  initialToleration?: PodToleration;
+  onSave: (toleration: PodToleration) => void;
+};
+
+const ManageTolerationModal: React.FC<ManageTolerationModalProps> = ({
+  isOpen,
+  onClose,
+  initialToleration,
+  onSave,
+}) => {
+  const [toleration, setToleration] = useState<PodToleration>(EMPTY_TOLERATION);
+
+  const isButtonDisabled = !toleration.key;
+
+  useEffect(() => {
+    if (initialToleration) {
+      setToleration(initialToleration);
+    }
+  }, [initialToleration]);
+
+  const handleUpdate = (updatedToleration: PodToleration) => {
+    setToleration(updatedToleration);
+  };
+
+  const onBeforeClose = () => {
+    setToleration(EMPTY_TOLERATION);
+  };
+
+  return (
+    <Modal
+      title="Manage Toleration"
```

Comment:
When editing the toleration, should we make the modal title something like `Edit toleration`?